src/hooks/screen/useMessageHook.ts

```diff
@@ -0,0 +1,132 @@
+import SimpleToast from 'react-native-simple-toast';
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Alert, NativeSyntheticEvent} from 'react-native';
+import {ContextMenuOnPressNativeEvent} from 'react-native-context-menu-view';
+import {useAnimatedStyle, useSharedValue} from 'react-native-reanimated';
+import {useCallback, useContext} from 'react';
+
+import AnonymousMessageRepo from '../../service/repo/anonymousMessageRepo';
+import ChatSchema from '../../database/schema/ChatSchema';
+import ShareUtils from '../../utils/share';
+import SignedMessageRepo from '../../service/repo/signedMessageRepo';
+import useChatUtilsHook from '../core/chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import {Context} from '../../context';
+import {DELETED_MESSAGE_TEXT} from '../../utils/constants';
+import {ReplyMessage, UseMessageHook} from '../../../types/hooks/screens/useMessageHook.types';
+import {setReplyTarget} from '../../context/actions/chat';
+
+function useMessageHook(): UseMessageHook {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const {selectedChannel} = useChatUtilsHook();
+  const [replyPreview, dispatch] = (useContext(Context) as unknown as any).chat;
+  const {anon_user_info_emoji_name} = selectedChannel?.rawJson?.channel || {};
+  const bubblePosition = useSharedValue(0);
+  const pulseAnimation = useSharedValue(1);
+
+  const setReplyPreview = useCallback((messageItem) => {
+    setReplyTarget(messageItem, dispatch);
+  }, []);
+
+  const clearReplyPreview = useCallback(() => {
+    setReplyTarget(null, dispatch);
+  }, []);
+
+  const deleteMessage = async (messageId: string, type: 'ANONYMOUS' | 'SIGNED', iteration = 0) => {
+    if (!localDb) return;
+
+    if (iteration > 5) {
+      SimpleToast.show("Can't delete message, please check your connection");
+      return;
+    }
+
+    try {
+      ChatSchema.updateDeletedChatType(localDb, messageId);
+      refresh('chat');
+
+      const {replyTarget} = replyPreview;
+      if (replyTarget && replyTarget?.id === messageId) {
+        const newReplyPreview = {...replyTarget};
+        newReplyPreview.message = DELETED_MESSAGE_TEXT;
+        newReplyPreview.message_type = 'deleted';
+        setReplyPreview(newReplyPreview);
+      }
+
+      if (type === 'ANONYMOUS') {
+        await AnonymousMessageRepo.deleteMessage(messageId);
+      } else {
+        await SignedMessageRepo.deleteMessage(messageId);
+      }
+    } catch (error) {
+      console.log('error on delete message:', error);
+      setTimeout(() => {
+        deleteMessage(messageId, type, iteration + 1).catch((e) => console.log(e));
+      }, 1000);
+    }
+  };
+
+  const getUserName = (item): string => {
+    if (item?.user?.username !== 'AnonymousUser') {
+      return item?.user?.username;
+    }
+    return `Anonymous ${anon_user_info_emoji_name}`;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getUserName` function could be simplified by using a ternary operator.

```diff
  const getUserName = (item): string => {
    if (item?.user?.username !== 'AnonymousUser') {
      return item?.user?.username;
    }
    return `Anonymous ${anon_user_info_emoji_name}`;
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->